loader/generated/vk_loader_extension_utils.h

```diff
@@ -0,0 +1,138 @@
+// Instance command lookup function
+VKAPI_ATTR void* VKAPI_CALL loader_lookup_instance_dispatch_table(const VkLayerInstanceDispatchTable *table, const char *name,
+                                                                  bool *found_name);
+
+// Device command lookup function
+VKAPI_ATTR void* VKAPI_CALL loader_lookup_device_dispatch_table(const VkLayerDispatchTable *table, const char *name);
+
+// Union of all known instance extensions that the loader may need to refer to when
+// making function pointer decisions.
+union loader_instance_extension_enables {
+    struct {
+        uint8_t khr_surface : 1;
+        uint8_t khr_swapchain : 1;
+        uint8_t khr_display : 1;
+        uint8_t khr_xlib_surface : 1;
+        uint8_t khr_xcb_surface : 1;
+        uint8_t khr_wayland_surface : 1;
+        uint8_t khr_win32_surface : 1;
+        uint8_t khr_video_queue : 1;
+        uint8_t khr_get_physical_device_properties2 : 1;
+        uint8_t khr_device_group_creation : 1;
+        uint8_t khr_external_memory_capabilities : 1;
+        uint8_t khr_external_semaphore_capabilities : 1;
+        uint8_t khr_external_fence_capabilities : 1;
+        uint8_t khr_performance_query : 1;
+        uint8_t khr_get_surface_capabilities2 : 1;
+        uint8_t khr_get_display_properties2 : 1;
+        uint8_t khr_fragment_shading_rate : 1;
+        uint8_t khr_surface_protected_capabilities : 1;
+        uint8_t ext_debug_report : 1;
+        uint8_t ggp_stream_descriptor_surface : 1;
+        uint8_t nv_external_memory_capabilities : 1;
+        uint8_t ext_validation_flags : 1;
+        uint8_t nn_vi_surface : 1;
+        uint8_t ext_direct_mode_display : 1;
+        uint8_t ext_acquire_xlib_display : 1;
+        uint8_t ext_display_surface_counter : 1;
+        uint8_t ext_swapchain_colorspace : 1;
+        uint8_t mvk_ios_surface : 1;
+        uint8_t mvk_macos_surface : 1;
+        uint8_t ext_debug_utils : 1;
+        uint8_t ext_sample_locations : 1;
+        uint8_t ext_calibrated_timestamps : 1;
+        uint8_t fuchsia_imagepipe_surface : 1;
+        uint8_t ext_metal_surface : 1;
+        uint8_t ext_tooling_info : 1;
+        uint8_t ext_validation_features : 1;
+        uint8_t nv_cooperative_matrix : 1;
+        uint8_t nv_coverage_reduction_mode : 1;
+        uint8_t ext_full_screen_exclusive : 1;
+        uint8_t ext_headless_surface : 1;
+        uint8_t ext_acquire_drm_display : 1;
+        uint8_t nv_acquire_winrt_display : 1;
+        uint8_t ext_directfb_surface : 1;
+        uint8_t qnx_screen_surface : 1;
+        uint8_t google_surfaceless_query : 1;
+    };
+    uint64_t padding[4];
+};
+
+// Union of all device extensions that the loader has to override at least one command for
+union loader_device_extension_enables {
+    struct {
+        uint8_t khr_swapchain : 1;
+        uint8_t khr_display_swapchain : 1;
+        uint8_t khr_device_group : 1;
+        uint8_t ext_debug_marker : 1;
+        uint8_t ext_debug_utils : 1;
```

Comment:
Yes, if you look it's always been that way because even though it's an instance extension, it also has device functionality.  So it's been treated as both (one of my greatest regrets was listening to people who said the two different extensions instance and device needed to be merged.  If you look at vk_loader_extension_utils.c in `extensions_create_device` you will see that the value is copied from the instance extension enable check as well.

Reply:
Ah, my mistake. I was reading this code change in isolation so that made me perk up.